components/tidb_query/Cargo.toml

```diff
@@ -39,8 +39,8 @@ safemem = { version = "0.3", default-features = false }
 flate2 = { version = "1.0", features = ["zlib"], default-features = false }
 slog = { version = "2.3", features = ["max_level_trace", "release_max_level_debug"] }
 slog-global = { version = "0.1", git = "https://github.com/breeswish/slog-global.git", rev = "0e23a5baff302a9d7bccd85f8f31e43339c2f2c1" }
-tipb = { git = "https://github.com/pingcap/tipb.git" }
-kvproto = { git = "https://github.com/pingcap/kvproto.git" }
+tipb = { git = "https://github.com/pingcap/tipb.git", default-features = false }
```

Comment:
I'm curious why it can compile as no features means no struct at all.

Reply:
It is because of your suggestion :-) The features are unioned together and in the top level Cargo.toml I set the Prost or proto features.

Reply:
Does it mean `tidb_query` can't be compiled alone?

Reply:
No, `cargo build -p tidb_query --features prost-codec` at the top-level will work. I believe this is because Cargo computes the whole dep graph including features and then only builds what is needed (I am not sure if this is a bug or a feature :-) )

Reply:
Get it. Seems features are required to get it built now.